build-tools/packages/build-tools/src/test/globPatterns.test.ts

```diff
@@ -0,0 +1,276 @@
+ *
+ * This function is used by LeafWithGlobInputOutputDoneFileTask.getFiles() to get input/output files
+ * for tasks with gitignore support. These tests verify the behavior that must be preserved when
+ * migrating from globby to tinyglobby.
+ *
+ * When migrating to tinyglobby, gitignore support must be implemented manually
+ * (e.g., using `git ls-files --others --ignored --exclude-standard` to get ignored files).
+ */
+describe("globWithGitignore (LeafTask file enumeration)", () => {
+	const gitIgnoredDir = path.join(globTestDataPath, "gitignored");
+	const gitIgnoredFile = path.join(gitIgnoredDir, "shouldBeIgnored.ts");
+
+	beforeEach(async () => {
+		// Create gitignored directory and file for testing
+		await mkdir(gitIgnoredDir, { recursive: true });
+		await writeFile(
+			gitIgnoredFile,
+			"/*!\n * Copyright (c) Microsoft Corporation and contributors. All rights reserved.\n * Licensed under the MIT License.\n */\n\n// This file should be ignored by gitignore\n",
+		);
+	});
+
+	afterEach(async () => {
+		// Clean up gitignored directory
+		await rm(gitIgnoredDir, { recursive: true, force: true });
+	});
+
+	it("includes gitignored files when gitignore option is false", async () => {
+		const results = await globWithGitignore(["**/*.ts"], {
+			cwd: globTestDataPath,
+			gitignore: false,
+		});
+		const filenames = results.map((f) => path.basename(f));
+		// Without gitignore filtering, should include the ignored file
+		assert(filenames.includes("shouldBeIgnored.ts"), "Should include gitignored files");
+		assert(filenames.includes("tracked.ts"), "Should include tracked files");
+	});
+
+	it("excludes gitignored files when gitignore option is true", async () => {
+		const results = await globWithGitignore(["**/*.ts"], {
+			cwd: globTestDataPath,
+			gitignore: true,
+		});
+		const filenames = results.map((f) => path.basename(f));
+		// With gitignore filtering, should exclude files matching .gitignore patterns
+		assert(!filenames.includes("shouldBeIgnored.ts"), "Should exclude gitignored files");
+		assert(filenames.includes("tracked.ts"), "Should include tracked files");
+	});
+
+	it("excludes gitignored files by default", async () => {
+		const results = await globWithGitignore(["**/*.ts"], {
+			cwd: globTestDataPath,
+		});
+		const filenames = results.map((f) => path.basename(f));
+		// Default behavior should exclude gitignored files
+		assert(
+			!filenames.includes("shouldBeIgnored.ts"),
+			"Should exclude gitignored files by default",
+		);
+		assert(filenames.includes("tracked.ts"), "Should include tracked files");
+	});
+
+	it("excludes files matching gitignore patterns", async () => {
+		const results = await globWithGitignore(["**/*"], {
+			cwd: globTestDataPath,
+			gitignore: true,
+		});
+		const filenames = results.map((f) => path.basename(f));
+		// The .gitignore contains "*.ignored" pattern
+		assert(
+			!filenames.includes("test.ignored"),
+			"Should exclude files matching *.ignored pattern",
+		);
```

Comment:
Nothing creates the `test.ignored` file as far as I can tell so this test passes without actually validating anything.